packages/teams-js/src/internal/communication.ts

```diff
@@ -766,20 +770,25 @@ function handleChildMessage(evt: DOMMessageEvent): void {
       // @ts-ignore
       sendMessageResponseToChild(message.id, message.uuid, Array.isArray(result) ? result : [result]);
     } else {
-      // No handler, proxy to parent
-      sendMessageToParent(
-        getApiVersionTag(ApiVersionNumber.V_2, ApiName.Tasks_StartTask),
-        message.func,
-        message.args,
-        (...args: any[]): void => {
-          if (Communication.childWindow) {
-            const isPartialResponse = args.pop();
-            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-            // @ts-ignore
-            sendMessageResponseToChild(message.id, message.uuid, args, isPartialResponse);
-          }
-        },
-      );
+      if (GlobalVars.allowMessageProxy || GlobalVars.webAuthWindowOpen) {
+        // No handler, proxy to parent
+        sendMessageToParent(
+          getApiVersionTag(ApiVersionNumber.V_2, ApiName.Tasks_StartTask),
+          message.func,
+          message.args,
+          (...args: any[]): void => {
+            if (Communication.childWindow) {
+              const isPartialResponse = args.pop();
+              // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+              // @ts-ignore
+              sendMessageResponseToChild(message.id, message.uuid, args, isPartialResponse);
+              runtime;
```

Comment:
This section of code takes the `func` and `args` from the message TeamsJS received from its child and sends them in a new post message up to the parent. It also sets up a handler for dealing with the response, passing the response back down to the child (using `sendMessageResponseToChild`) instead of handling the returned data itself.